src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,169 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.RegistrationServiceImplException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+
+    @BeforeAll
+    static void initDB() {
+        Storage.people.add(new User("ValidUser01", "valid_password01", 18));
+        Storage.people.add(new User("ValidUser02", "valid_password02", 25));
+        Storage.people.add(new User("ValidUser03", "valid_password03", 40));
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @AfterAll
+    static void disconnectFromDB() {
+        registrationService = null;
+    }
+
+    @Test
+    void register_userNullValue_notOk() {
+        assertThrows(RegistrationServiceImplException.class,
+                () -> registrationService.register(null));
+    }
+
+    @Test
+    void register_userExists_notOk() {
+        assertThrows(RegistrationServiceImplException.class,
+                () -> registrationService.register(new User(
+                        "ValidUser01", "valid_password01", 18)));
+    }
+
+    @Test
+    void register_loginNullValue_notOk() {
+        assertThrows(RegistrationServiceImplException.class,
+                () -> registrationService.register(new User(null, "123456", 18)));
+    }
+
+    @Test
+    void register_loginShort_notOk() {
+        assertThrows(RegistrationServiceImplException.class,
+                () -> registrationService.register(new User("", "123456", 18)));
+        assertThrows(RegistrationServiceImplException.class,
+                () -> registrationService.register(new User("Ada", "123456", 18)));
+        assertThrows(RegistrationServiceImplException.class,
+                () -> registrationService.register(new User("Adam", "123456", 18)));
+        assertThrows(RegistrationServiceImplException.class,
+                () -> registrationService.register(new User("Adame", "123456", 18)));
+    }
+
+    @Test
+    void register_loginExactlyMinLen_edgeCase_Ok() throws RegistrationServiceImplException {
```

Comment:
remove throws from all test cases
```suggestion
    void register_loginExactlyMinLen_edgeCase_Ok() {
```